File: 프로그래머스/기초 트레이닝 - Level 0/문자열 바꿔서 출력하기-solve.java
class Solution {
    public int solution(String myString, String pat) {
        StringBuilder sb = new StringBuilder();
        
        for (String str : myString.split("")) {
            if (str.equals("A")) {
                sb.append("B");
            } else if (str.equals("B")) {
                sb.append("A");
            } else {
                sb.append(str); 
            }
        }
        return sb.toString().contains(pat) ? 1 : 0; 
    }
}

File: 프로그래머스/기초 트레이닝 - Level 0/문자열 바꿔서 출력하기-solve.py
def solution(myString, pat):
    changed_string = ""
    for alpha in myString:
        if alpha == "A":
            changed_string += "B"
        elif alpha == "B":
            changed_string += "A"
        else:
            changed_string += alpha 
            
    return 1 if pat in changed_string else 0